### PR TITLE
Audit API Docs

### PR DESCRIPTION
**Pull request by riastrad**
## Context

🍷 pairs well with #1345

This PR updates the documentation to remove mention of functionality that is no longer supported post the EOL of the Legacy Maps API. This includes removing documentation of `L.gridLayer` & `L.gridControl`. 

cc/ @mapbox/docs @mapbox/static-apis

### CODE REVIEW

File: docs/_posts/api/v3.3.1/0200-01-01-l-mapbox-map.html [hunk 1]
```diff
@@ -25,7 +25,7 @@ <h2 id="section-l-mapbox-map">L.mapbox.map(element, id|url|tilejson, options)</h
...
 <td><strong>string</strong> if <em>id</em> or <em>url</em> <strong>object</strong> if <em>tilejson</em></td>
-<td>url can be <ul><li>A map <code>id</code> string <code>examples.map-foo</code></li><li>A comma separated list of map <code>id</code> strings <code>examples.map-foo,examples.map-bar</code> <a href="https://www.mapbox.com/mapbox.js/example/v1.0.0/compositing/">example</a></li><li>A URL to TileJSON, like <code>{{site.tileApi}}/v3/mapbox.dark.json</code></li><li>A <a href="https://docs.mapbox.com/api/maps/#retrieve-tilejson-metadata">TileJSON</a> object, from your own Javascript code</li></ul></td>
+<td>url can be <ul><li>A map <code>id</code> string <code>examples.map-foo</code></li><li>A comma separated list of map <code>id</code> strings <code>examples.map-foo,examples.map-bar</code> <a href="https://www.mapbox.com/mapbox.js/example/v1.0.0/compositing/">example</a></li><li>A URL to TileJSON, like <code>{{site.tileApi}}/v3/mapbox.satellite.json</code></li><li>A <a href="https://docs.mapbox.com/api/maps/#retrieve-tilejson-metadata">TileJSON</a> object, from your own Javascript code</li></ul></td>
 </tr>
 <tr>
 <td>options</td>
```
Comment (katydecorah) [inline docs/_posts/api/v3.3.1/0200-01-01-l-mapbox-map.html:31]: Line 33 contains two links to L.mapbox.gridLayer, should those be removed, too?

**Review by katydecorah — COMMENTED** [review]
@riastrad looking good! In addition to the line comment, we'll also want to remove `L.mapbox.gridLayer` and `L.mapbox.gridControl` from the `navigation` property in the frontmatter of `0200-01-01-v3.3.1.html` this will remove the pages from the left sidebar.

File: docs/_posts/api/v3.3.1/0200-01-01-l-mapbox-map.html (hunk 1 above)
Comment (riastrad) [reply docs/_posts/api/v3.3.1/0200-01-01-l-mapbox-map.html:31]: 👍 great catch 8dd17dde

**Comment by riastrad** [comment]
Thanks for the review, @katydecorah. After cfd21cb8, we should be all set for a final review before merging.

**Review by katydecorah — APPROVED** [review]
Thanks @riastrad!

There are two links for l-mapbox-girdcontrol that I missed in my first review that we may want to remove as well in: `0200-01-01-l-mapbox-sanitize.html` and `0200-01-01-l-mapbox-template.html`, but I'll approve in advance since these changes are small.

**Comment by riastrad** [comment]
Thanks very much, @katydecorah ❤️

**Pull request merged**